spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java

```diff
@@ -0,0 +1,283 @@
+
+	private boolean enableEmptyPayload = false;
+
+	private boolean enableBatch = false;
+
+	private ThreadFactory threadFactory;
+
+	/**
+	 * Create new Debezium message producer inbound channel adapter.
+	 * @param debeziumBuilder - pre-configured Debezium Engine Builder instance.
+	 */
+	public DebeziumMessageProducer(Builder<ChangeEvent<byte[], byte[]>> debeziumBuilder) {
+		Assert.notNull(debeziumBuilder, "The Debezium Engine Builder is null!");
+		this.debeziumEngineBuilder = debeziumBuilder;
+	}
+
+	/**
+	 * Defines if a single or a batch of messages are send downstream.
+	 * @param enableBatch False - sends downstream {@link Message} for every {@link ChangeEvent data change event} read
+	 * from the source database. True - sends the received {@link List} of {@link ChangeEvent}s as a raw {@link Message}
+	 * payload in a single (batch) downstream message. Such payload is not serializable and would required custom
+	 * serialization/deserialization implementation.
+	 */
+	public void setEnableBatch(boolean enableBatch) {
+		this.enableBatch = enableBatch;
+	}
+
+	/**
+	 * Enables support for tombstone (aka delete) messages. On a database row delete, Debezium can send a tombstone
+	 * change event that has the same key as the deleted row and a value of {@link Optional.empty}. This record is a
+	 * marker for downstream processors. It indicates that log compaction can remove all records that have this key.
+	 * When the tombstone functionality is enabled in the Debezium connector configuration you should enable the empty
+	 * payload as well.
+	 * @param enableEmptyPayload True enables the empty payload handling mechanism. False by default.
+	 */
+	public void setEnableEmptyPayload(boolean enableEmptyPayload) {
+		this.enableEmptyPayload = enableEmptyPayload;
+	}
+
+	/**
+	 * Set a {@link ThreadFactory} for Debezium executor. Defaults to the {@link CustomizableThreadFactory} based on a
+	 * {@code debezium-} prefix.
+	 * @param threadFactory the {@link ThreadFactory} to use.
+	 */
+	public void setThreadFactory(ThreadFactory threadFactory) {
+		Assert.notNull(threadFactory, "'threadFactory' must not be null");
+		this.threadFactory = threadFactory;
+	}
+
+	/**
+	 * Outbound message content type. Should be aligned with the {@link SerializationFormat} configured for the
+	 * {@link DebeziumEngine}.
+	 */
+	public void setContentType(String contentType) {
+		Assert.hasText(contentType, "Invalid content type: " + contentType);
+		this.contentType = contentType;
+	}
+
+	/**
+	 * Specifies how to convert Debezium change event headers into Message headers.
+	 * @param headerMapper concrete HeaderMapping implementation.
+	 */
+	public void setHeaderMapper(HeaderMapper<List<Header<Object>>> headerMapper) {
+		Assert.notNull(headerMapper, "'headerMapper' must not be null.");
+		this.headerMapper = headerMapper;
+	}
+
+	/**
+	 * @return Returns current header mapper.
+	 */
+	public HeaderMapper<List<Header<Object>>> getHeaderMapper() {
```

Comment:
Why do we need this getter?
It does not give any gain from a management perspective and better to design our classes as immutable as possible.

Reply:
because it is handy. check the integration tests

Reply:
Then use `TestUtils.getProperty()`.
Whatever is not intended for end-user must be hidden.
It is really not so handy given an outcome that channel adapter might be mutated.
So, we may got some GH issue with a strange behavior.
And that will be exactly our fault since we have exposed this property.

Reply:
What I meant it that this is usefulul to simplify the debezium-message-producer configuration. Given that later is configured by default with DefaultDebeziumHeaderMapper, e.g. the instance is already there, it is awkward to expect someone to create a new instance from the same class just be able to configure it slitely differntly. 
This was the primary reason I introduced the mirror setAllowedHeaderNames().

Reply:
Well, for consistency with Java Beans spec it is better to keep this as straightforward as possible.
The `setAllowedHeaderNames()` is really nor this component responsibility and even if we inject a new instance of the same class, we still just override its internal value.
The setter way is easy to follow from an XML config and so on.
It is really better to give only one choice to end-users: "want some heads to filter out? - Provide a respective `HeaderMapper`".

When we design this component we need to think about responsibility level and convenience for end-users.
The getter is not so convenient and does not give a clue what we support to do with its result in the target project.
Users of our API really don't care how is that convenient for your tests.

When we introduce Java DSL for this channel adapter, that's where we will expose a convenient `headersToMap()` option.

Reply:
> Users of our API really don't care how is that convenient for your tests.
As if explained, if just someone have read it, It is convenient for the users not for my tests. Anyway feel free to change it as it find it fit

Reply:
Well, this is your PR and it is hard for me to change something especially if you already tie into it in the test.
Moreover this is just a discussion without any pressure.
I just don't feel comfortable to expose a getter for end-user configuration considerations, where existing setter is really just a single place of truth and it should ensure an immutability state of our component after configuration phase.